girder/utility/s3_assetstore_adapter.py

```diff
@@ -727,45 +575,33 @@ def makeBotoConnectParams(accessKeyId, secret, service=None):
 
     :param accessKeyId: the S3 access key ID
     :param secret: the S3 secret key
-    :param service: the name of the service in the form
-                    [http[s]://](host domain)[:(port)].
+    :param service: alternate service URL
     :returns: boto connection parameter dictionary.
     """
     if accessKeyId and secret:
-        connect = {
+        params = {
             'aws_access_key_id': accessKeyId,
             'aws_secret_access_key': secret,
-            }
+            'config': botocore.client.Config(signature_version='s3v4')
+        }
     else:
-        connect = {
-            'anon': True
+        params = {
+            'config': botocore.client.Config(signature_version=botocore.UNSIGNED)
         }
 
     if service:
-        service = re.match("^((https?)://)?([^:/]+)(:([0-9]+))?$", service)
-        if service.groups()[1] == 'http':
-            connect['is_secure'] = False
-        connect['host'] = service.groups()[2]
-        if service.groups()[4] is not None:
-            connect['port'] = int(service.groups()[4])
-    else:
-        # If we are using sigv4, we MUST specify a host for boto.  If no
-        # service is specified by the user, use the default used in boto
-        connect['host'] = boto.config.get('s3', 'host', 's3.amazonaws.com')
-    return connect
+        serviceRe = re.match('^((https?)://)?([^:/]+)(:([0-9]+))?$', service)
+        if serviceRe.groups()[1] == 'http':
+            params['use_ssl'] = False
+        params['endpoint_url'] = service
+
+    # TODO(zach) region parameter? Might not be necessary
+    return params
 
 
 def _deleteFileImpl(event):
-    """
-    Uses boto to delete the key.
-    """
-    info = event.info
-    conn = botoConnectS3(info.get('botoConnect', {}))
-    bucket = conn.lookup(bucket_name=info['bucket'], validate=False)
-    key = bucket.get_key(info['key'], validate=True)
-    if key:
-        bucket.delete_key(key)
+    _botoS3(event.info['connectParams']).delete_object(
+        Bucket=event.info['bucket'], Key=event.info['key'])
 
 
-events.bind('_s3_assetstore_delete_file', '_s3_assetstore_delete_file',
-            _deleteFileImpl)
+events.bind('_s3_assetstore_delete_file', '_s3_assetstore_delete_file', _deleteFileImpl)
```

Comment:
I'm running into scenarios where my file isn't being deleted on S3, is this a matter of their eventually consistent model?